presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -354,11 +353,9 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                 List<Node> stageNodeList;
                 if (plan.getSubStages().isEmpty()) {
                     // no remote source
-                    boolean preferDynamic = groupedExecutionForStage && isDynamicSchduleForGroupedExecution(session);
-                    bucketNodeMap = nodePartitioningManager.getBucketNodeMap(session, partitioningHandle, preferDynamic);
-                    if (bucketNodeMap.isDynamic()) {
-                        verify(preferDynamic);
-                    }
+                    boolean dynamicLifespanSchedule = plan.getFragment().getStageExecutionDescriptor().isDynamicLifespanSchedule();
+                    bucketNodeMap = nodePartitioningManager.getBucketNodeMap(session, partitioningHandle, dynamicLifespanSchedule);
+                    verify(bucketNodeMap.isDynamic() == dynamicLifespanSchedule);
```

Comment:
You might also want to verify it is not dynamic in the `else` branch in case something goes wrong in the planner?

Reply:
Please add a message in case it ever fails. Also maybe leave a comment explaining why this condition has to always be true (explain that it is checked in the planner, so the `dynamicLifespanSchedule` is set to the `bucketNodeMap.isDynamic()` in planner).

Reply:
@arhimondr : Thanks. I prefer either adding the comment or the message, but not both . In this case, I will add comment. The rational is verification error is a very severe logic error, and we don't need to show this to end-user.